crates/subspace-networking/src/node_runner.rs

```diff
@@ -213,34 +213,27 @@ where
         }
     }
 
-    async fn handle_peer_dialing(&mut self) {
-        let local_peer_id = *self.swarm.local_peer_id();
-        let connected_peers = self.swarm.connected_peers().cloned().collect::<Vec<_>>();
+    // Handle DSN online status signaling
+    fn signal_online_status(&mut self) {
+        let current_online_status = self.swarm.connected_peers().next().is_some();
+        let previous_online_status = *self.online_status_observer_tx.borrow();
 
-        // Handle DSN connection signaling
-        // TODO: consider moving it a separate watcher with decoupled check period.
-        let current_dsn_connection_state = !connected_peers.is_empty();
-        let previous_dsn_connection_state = {
-            if let Some(shared) = self.shared_weak.upgrade() {
-                *shared.dsn_connection_observer_rx.borrow()
-            } else {
-                return; // App is closing
-            }
-        };
-        if previous_dsn_connection_state != current_dsn_connection_state {
-            if let Err(err) = self
-                .dsn_connection_observer_tx
-                .send(current_dsn_connection_state)
-            {
+        if previous_online_status != current_online_status {
+            if let Err(err) = self.online_status_observer_tx.send(current_online_status) {
                 error!("DSN connection observer channel failed: {err}")
             }
 
-            if current_dsn_connection_state {
+            if current_online_status {
                 info!("DSN connection established.");
             } else {
                 warn!("DSN connection lost.");
             }
```

Comment:
This is not for library level to log though

Reply:
`online_status_observer` is already public. Want to see it in the app - write corresponding code in the app. As a general rule of thumb library should not print anything with `info` level, application should. It can print serious errors, but should strongly prefer to return them as `Result` instead too.

So 1 is one of those awkward decisions I want to not see in the codebase and 2 is already offered by `online_status_observer`, so I'm not sure what subscription you're talking about here.